omp/__init__.py

```diff
@@ -87,29 +93,31 @@ def init_not_msvc(self):
             except (OSError, CalledProcessError):
                 pass
 
-        # Try to load find libgomp shared library using loader search dirs
-        libgomp_path = find_library("gomp")
-
-        # Try to use custom paths if lookup failed
-        for path in paths:
-            if libgomp_path:
-                break
-            path = path.strip()
-            if os.path.isdir(path):
-                libgomp_path = find_library(os.path.join(str(path), "libgomp"))
-
-        if not libgomp_path:
-            raise ImportError("I can't find a shared library for libgomp,"
-                              " you may need to install it or adjust the "
-                              "LD_LIBRARY_PATH environment variable.")
-        else:
-            # Load the library
-            try:
-                self.libomp = ctypes.CDLL(libgomp_path)
-            except OSError:
-                raise ImportError("found openMP library '{}' but couldn't load it. "
-                                  "This may happen if you are cross-compiling.".format(libgomp_path))
-            self.version = 45
+        for libomp_name in libomp_names:
+            # Try to load find libomp shared library using loader search dirs
+            libomp_path = find_library(libomp_name)
+
+            # Try to use custom paths if lookup failed
+            for path in paths:
+                if libomp_path:
+                    break
+                path = path.strip()
+                if os.path.isfile(os.path.join(path, libomp_name)):
+                    libomp_path = os.path.join(path, libomp_name)
+
+            if libomp_path:
+                # Load the library
+                try:
+                    self.libomp = ctypes.CDLL(libomp_path)
+                except OSError:
+                    raise ImportError("found openMP library '{}' but couldn't load it. "
+                                      "This may happen if you are cross-compiling.".format(libomp_path))
+                self.version = 45
+                return
+
+        raise ImportError("I can't find a shared library for libomp,"
+                          " you may need to install it or adjust the "
+                          "LD_LIBRARY_PATH environment variable.")
```

Comment:
this last hint is incorrect on OSX. would that be possible to store the relevant env variable in a list that would be set to `['LD_LIBRARY_PATH']` or `['DYLD_LIBRARY_PATH', 'DYLD_FALLBACK_LIBRARY_PATH']` based on the platform, and use that list for both `paths` initialisation and `ImportError` message?